lms/services/oauth_http.py

```diff
@@ -122,12 +126,15 @@ def _token_request(self, token_url, data, auth):
             validated_data["access_token"],
             validated_data.get("refresh_token"),
             validated_data.get("expires_in"),
+            service=self.service,
         )
 
         return validated_data["access_token"]
 
 
-def factory(_context, request):
+def factory(_context, request, service: Service = Service.LMS) -> OAuthHTTPService:
```

Comment:
This allows constructing an `OAuthHTTPService` with a non-default API service. We don't have any other factory methods that are ever called directly, as opposed to only being called by `pyramid_services`.